web/src/main/webapp/app/shared/treemultiselector/treeMultiSelectorUtil.js

```diff
@@ -24,24 +24,97 @@ angular.module('StatisticsApp.treeMultiSelector.util', ['underscore'])
+          allSelected = allSelected && (externalApi.isCareUnitWithNoUnits(sub) ? true : sub.allSelected) &&
+              (externalApi.isCareUnit(sub) ? sub.isSelected : true);
         });
-        if (allSelected) {
-          item.allSelected = true;
-          item.someSelected = false;
-        } else {
-          item.allSelected = false;
-          item.someSelected = !!someSelected;
+        if (!externalApi.isDistributedCareUnit(item)) {
+          if (allSelected) {
+            item.allSelected = true;
+            item.someSelected = false;
+          } else {
+            if (item.isSelected && !allSelected) {
+              item.isSelected = false;
+            }
+            item.allSelected = false;
+            item.someSelected = !!someSelected;
+          }
         }
       }
     };
 
-    return externalApi;
+    externalApi.updateCareUnits = function(item) {
+      if (!item.subs || item.subs.length === 0 || !externalApi.isCounty(item.subs[0])) {
+        return;
+      }
+
+      var distributedCareUnits = externalApi.getDistributedCareUnits(item);
+      var sortedDistributedUnits = externalApi.getDistributedUnits(distributedCareUnits);
 
+      _.each(distributedCareUnits, function(careUnit) {
+        var someSelected = false;
+        var allSelected = true;
+        _.each(_.find(sortedDistributedUnits, {careUnitId: careUnit.id}).units, function(unit) {
+          someSelected = someSelected || unit.someSelected || unit.allSelected;
+          allSelected = allSelected && unit.allSelected;
+        });
+
+        careUnit.allSelected = allSelected;
+        careUnit.someSelected = allSelected ? false : someSelected;
+        if (careUnit.isSelected && !allSelected) {
+          careUnit.isSelected = false;
+        }
+      });
+    };
+
+    externalApi.getDistributedCareUnits = function(item) {
+      var distributedCareUnits = [];
+      _.each(item.subs, function(county) {
+        _.each(county.subs, function(munip) {
+          distributedCareUnits =  _.concat(distributedCareUnits, _.filter(munip.subs, {distributedCareUnit: true}));
+        });
+      });
+      return distributedCareUnits;
+    };
+
+    externalApi.getDistributedUnits = function(distributedCareUnits) {
+      var sortedDistributedUnits = [];
+      _.each(distributedCareUnits, function(careUnit) {
+        var existingCareUnit = _.find(sortedDistributedUnits, {careUnitId: careUnit.id});
+        if (existingCareUnit) {
+          existingCareUnit.units = _.concat(existingCareUnit.units, careUnit.subs);
+        } else {
+          sortedDistributedUnits.push({careUnitId: careUnit.id, units: careUnit.subs});
+        }
+      });
+      return sortedDistributedUnits;
+    };
+
+    externalApi.isCareUnit = function(item) {
+      return item.numericalId && typeof item.numericalId === 'string' && item.numericalId.indexOf('vardenhet') > -1;
```

Comment:
Is it possible to create a constant for 'vardenhet' and share between treeMultiSelectorUtil.js and submenu.directive.js?